src/systems/systems/src/hydraulic/update_iterator.rs

```diff
@@ -0,0 +1,288 @@
+use crate::simulation::UpdateContext;
+use std::time::Duration;
+
+/// This is a structure that will handle a fixed step like update.
+/// When update() is called, it will build an array of frame updates to do, all having the same fixed duration
+/// If all those fixed steps added do not sum up as the whole update time, a time lag accumulator is incremented so we catch up
+/// with real time at next updates.
+///
+/// Usage:
+/// Call update() at each sim updtate
+/// Then a for loop over this as an iterator will return one frame per loop with corresponding fixed step duration
```

Comment:
**thought**: I find this description to be very technical. It describes the implementation and after reading it I still have a hard time understanding what this type is supposed to do.

```suggestion
/// Provides fixed time interval looping.
///
/// ## Example scenario
/// With a fixed time interval of 10 ms and a frame delta of 35 ms, this type will provide three
/// iterations of 10 ms. The remaining 5 ms will carry over to the next frame. When the next frame
/// takes 15 ms, this will result in 2 iterations of 10 ms (15 ms + 5 ms).
```